pkg/services/cleanup/cleanup.go

```diff
@@ -67,7 +68,7 @@ func (srv *CleanUpService) Run(ctx context.Context) error {
 func (srv *CleanUpService) cleanUpOldAnnotations(ctx context.Context) {
 	cleaner := annotations.GetAnnotationCleaner()
 	affected, affectedTags, err := cleaner.CleanAnnotations(ctx, srv.Cfg)
-	if err != nil {
+	if err != nil && !errors.Is(err, context.DeadlineExceeded) {
```

Comment:
```suggestion
	if !errors.Is(err, context.DeadlineExceeded) {
```

Just tried this in the go playground, we don't need to check `if err != nil` before checking `errors.Is`

https://play.golang.org/p/kSzOvEdbyLB

Minor nit tho. Looks good to me :D

Reply:
Also, this actually changes the logic, so cancel that.